src/components/map/use-map-instance.ts

```diff
@@ -35,7 +35,9 @@ export function useMapInstance(
   const {
     id,
     defaultBounds,
+    // @ts-expect-error TS complains that it's not defined in MapProps
     defaultCenter,
```

Comment:
Why TS thinks defaultCenter is not defined here? ... when testing on pojo typed as MapProps that prop will be there, as expected.

Reply:
I recreated a simplified version of the situation  [in ts-playground](https://www.typescriptlang.org/play?#code/C4TwDgpgBAMghsGA7A5gXigbwDYIFxICuAtgEYQBOANFNqnlEWZQL4DcAUB6JFALJwwAeTDAAlgHskAZygZMHKFADGEJMEoB+BvESoO7Lj2gCwAYTjFKcAAoUJYWfIBmEiQybkKLKADIoABSKWKrqlDoIyCg+AD5YACYQznCE2MBmahoUEXrRHACUnNzgJoJ2Dk78giLiUrL+phZWFLb2jkXOhEjKtUhQzgFgDKbljvlYwcp1wAlJKWkZYd5yUGCcLEA). 

The error message _"Property 'defaultCenter' does not exist on type 'MapProps'."_, is a bit misleading, but its true: You're using union-types, and those only contain the fields that are present in all variations of the union (in case of `{center: X} | {defaultCenter: X}` that is the empty object). The center prop works since it's already defined in the `google.maps.MapOptions` type. 

What would work here is to specify the union as `{center: LatLng, defaultCenter?:LatLng} | {center?: LatLng, defaultCenter: LatLng}`, [see here](https://www.typescriptlang.org/play?ssl=20&ssc=1&pln=20&pc=13#code/C4TwDgpgBAMghsGA7A5gXigbwDYIFxICuAtgEYQBOANFNqnlEWZQL4DcAUB6JFALJwwAeTDAAlgHskAZygZMHKFABmEiQH4GTchU5KAxhCTBKm2AmQoO7Lj2gCwAYTjFKcAAoUJYWRgAUiliGxpQM8IioNAAmEMpwhNjAjkYmFJrhlixQAD5BKaZ4GZFQMXEJSfkUYRaoLBwAlJzc4PaCnt6+-IIi4lKyAGRdTi5u7T5NyoRI+r1IKn5gDA5j0vVYgfp9wHkh1CWx8YnJu1kYYJx1HMp+mMGpDDj4AIw0dCh4TywsjVc3pYcVXYPXDAD6veifb6ca6YVQSBifRpAA). The union type then resolves to `{center?:LatLng, defaultCenter?:LatLng}`.

Reply:
[Here's an alternative](https://www.typescriptlang.org/play?ssl=26&ssc=1&pln=26&pc=13#code/C4TwDgpgBAMghsGA7A5gXigbwDYIFxICuAtgEYQBOANFNqnlEWZQL4DcAUF6JFALJwwAeTDAAlgHskAZygZMHKFABmEiQH4GTchU5KAxhCTBKm2AmQoO7buGgBhKcAoTs2CABN7cYpTgAFFzBZeUNjSgZ4RFQaDwhlOEJsYHsjEwozJAgAN1ZOHmgAVSR9Jxc3T29fCgCgkKw4hKSUtIjzaJQaMPTMnLyuDgL+QSq-QIlguSgACkUoR2Ny9y8fMbqoAB8oYtLF12XRmvHgjgBKTkG7YbBj+oFhUUkZKAAya8PaiekLjmVCEvEUhU0zADHut1OWDmu2kwCw3UosXiiWSqXCFBYUzAnBYXGU00wCIoDBw+AAjDQ6Cg8GSWCxzr8CY0US10STcMAaZT6LT6Zx8ZhVBIGLTzkA) that produces slightly more readable error-messages.

Reply:
Seems it's a puzzle if TS can report this error as "either center or defaultCenter has to be provided", so far warnings had only one of two prop names. This and unsolved @ts-expect-error ... I will switch to console.warn solution.

Reply:
I think we can go with the [last option](https://github.com/visgl/react-google-maps/pull/308#discussion_r1560848530), I like the Idea of being able to statically warn about missing required props, even if the error-message is a bit clunky. I'll try a few more things to see if it gets any better...